notify.go

```diff
@@ -18,25 +18,10 @@ func (as *AmbientAutoNAT) OpenedStream(net network.Network, s network.Stream) {}
 func (as *AmbientAutoNAT) ClosedStream(net network.Network, s network.Stream) {}
 
 func (as *AmbientAutoNAT) Connected(net network.Network, c network.Conn) {
-	p := c.RemotePeer()
-
-	go func() {
-		// add some delay for identify
-		time.Sleep(AutoNATIdentifyDelay)
-
-		protos, err := as.host.Peerstore().SupportsProtocols(p, AutoNATProto)
-		if err != nil {
-			log.Debugf("error retrieving supported protocols for peer %s: %s", p, err)
-			return
-		}
-
-		if len(protos) > 0 {
-			log.Infof("Discovered AutoNAT peer %s", p.Pretty())
-			as.mx.Lock()
-			as.peers[p] = as.host.Peerstore().Addrs(p)
-			as.mx.Unlock()
-		}
-	}()
+	select {
+	case as.candidatePeers <- c:
```

Comment:
We can actually just check if the connection is inbound here to pre-filter (and rename this to `inboundConns`). Or maybe we should just check everything here?

Reply:
We do want to move the probe time forward when we detect an inbound conn. So, even if check everything here, we'd still want to get this inbound conn event to the event loop in `background()`.